src/stream/src/executor/sink.rs

```diff
@@ -50,30 +53,72 @@ impl<S: StateStore> SinkExecutor<S> {
             _store,
             properties,
             identity: format!("SinkExecutor_{:?}", executor_id),
+            pk_indices: Default::default(), // todo
         }
     }
 
     #[try_stream(ok = Message, error = StreamExecutorError)]
     async fn execute_inner(self) {
         let sink_config = SinkConfig::from_hashmap(self.properties.clone())
             .map_err(StreamExecutorError::sink_error)?;
-        let _sink = build_sink(sink_config);
 
-        // TODO(tabVersion): the flag is required because kafka transaction requires at least one
+        let mut sink = build_sink(sink_config).map_err(StreamExecutorError::sink_error)?;
+
+        // the flag is required because kafka transaction requires at least one
         // message, so we should abort the transaction if the flag is true.
-        #[allow(clippy::no_effect_underscore_binding)]
-        let _empty_epoch_flag = true;
+        let mut empty_epoch_flag = true;
+        let mut in_transaction = false;
```

Comment:
Will this be removed after we adopt the new transaction interfaces? cc @tabVersion